lib/extensions/ar_miq_set.rb

```diff
@@ -37,38 +27,51 @@ def make_memberof(set)
   end
 end # module ActsAsMiqSetMember
 
+module ActiveRecord
+  class Base
+    def self.acts_as_miq_set_member
+      include ActsAsMiqSetMember
+    end
+
+    def self.acts_as_miq_set(model_class = nil)
+      include ActsAsMiqSet
+
+      self.model_class = model_class unless model_class.nil?
+    end
+  end
+end
+
 module ActsAsMiqSet
   extend ActiveSupport::Concern
   included do
-    include RelationshipMixin
-    self.default_relationship_type ||= "membership"
-
     include UuidMixin
 
     self.table_name         = "miq_sets"
     self.inheritance_column = :set_type
 
     serialize :set_data
 
-    validates_presence_of     :name
-    validates_uniqueness_of   :name,
-                              :scope => [:set_type, :userid, :group_id],
-                              :if    => proc { |c| c.class.in_my_region.exists?(:name => c.name) }
-    validates_presence_of     :description
+    validates :name,
+              :presence                => true,
+              :uniqueness_when_changed => [:set_type, :userid, :group_id],
+              :if                      => proc { |c| c.class.in_my_region.exists?(:name => c.name) }
```

Comment:
total total tangent:

Having trouble understanding this. Is this a `:unique_within_region`?

```suggestion
    validates :name,
              :presence             => true,
              :unique_within_region => [:set_type, :userid, :group_id],
```

Reply:
That was pulled from Jason's change when it was part of `app/models/miq_set.rb`:

https://github.com/ManageIQ/manageiq/pull/21227/files#diff-27a8e11ee8322ede82d0e9273cb9f6e059ef90a5bc50cea38f3fe1ab585ba5ccR10
https://github.com/ManageIQ/manageiq/pull/20665/files#diff-27a8e11ee8322ede82d0e9273cb9f6e059ef90a5bc50cea38f3fe1ab585ba5ccR11

(note, those are two different PRs)

Reply:
I think this is a mistake, as I now have a change that puts it back to how it was.

Not sure if the suggested change would work the same, but how it was seems to be "more correct" than what it was changed to in the code I inherited.  Will test with your version (I assume just a more terse form) tomorrow.